checker/src/visitors.rs

```diff
@@ -465,18 +469,25 @@ impl<'analysis, 'compilation, 'tcx, E> MirVisitor<'analysis, 'compilation, 'tcx,
             {
                 return devirtualized_summary;
             }
-            self.assume_function_is_angelic = true;
+            // Assume the function as angelic unless we have set the flaky option.
+            self.assume_function_is_angelic = !self.options.flaky;
```

Comment:
Hah. Not flaky == Angelic. That is a low bar. :-)

Sound works. If you must, "demonic".